app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +135,223 @@ object DialogUtils {
+                        .toIntOrNull() != null && satText.text.toString().toIntOrNull() != null
+                ) {
+                    val calculatedColorState = ColorUtils.calculateColorWithHsv(
+                        floatArrayOf(
+                            hueText.text.toString().toFloat().coerceIn(0f, 360f),
+                            (satText.text.toString().toFloat() / 100f).coerceIn(0f, 1f),
+                            (it.toString().toFloat() / 100f).coerceIn(0f, 1f)
+                        )
+                    )
+                    forceViewUpdates(calculatedColorState)
+                }
+            }
+            redText.addTextChangedListener {
+                if (it.toString().toIntOrNull() != null && greenText.text.toString()
+                        .toIntOrNull() != null && blueText.text.toString().toIntOrNull() != null
+                ) {
+                    val calculatedColorState = ColorUtils.calculateColorWithRgb(
+                        intArrayOf(
+                            it.toString().toInt().coerceIn(0, 255),
+                            greenText.text.toString().toInt().coerceIn(0, 255),
+                            blueText.text.toString().toInt().coerceIn(0, 255)
+                        )
+                    )
+                    forceViewUpdates(calculatedColorState)
+                }
+            }
+            greenText.addTextChangedListener {
+                if (it.toString().toIntOrNull() != null && redText.text.toString()
+                        .toIntOrNull() != null && blueText.text.toString().toIntOrNull() != null
+                ) {
+                    val calculatedColorState = ColorUtils.calculateColorWithRgb(
+                        intArrayOf(
+                            redText.text.toString().toInt().coerceIn(0, 255),
+                            it.toString().toInt().coerceIn(0, 255),
+                            blueText.text.toString().toInt().coerceIn(0, 255)
+                        )
+                    )
+                    forceViewUpdates(calculatedColorState)
+                }
+            }
+            blueText.addTextChangedListener {
+                if (it.toString().toIntOrNull() != null && greenText.text.toString()
+                        .toIntOrNull() != null && redText.text.toString().toIntOrNull() != null
+                ) {
+                    val calculatedColorState = ColorUtils.calculateColorWithRgb(
+                        intArrayOf(
+                            redText.text.toString().toInt().coerceIn(0, 255),
+                            greenText.text.toString().toInt().coerceIn(0, 255),
+                            it.toString().toInt().coerceIn(0, 255)
+                        )
+                    )
+                    forceViewUpdates(calculatedColorState)
+                }
+            }
+            sheet.setContentView(view)
+            val behavior = BottomSheetBehavior.from(view.parent as View)
```

Comment:
Why hold this as a variable if you're only using it for one call?

Reply:
Now it is being used for more than one call to toggle whether the sheet should be draggable or not, so I am going to keep this as a variable for now.